ci/ReleaseTool/Common.cs

```diff
@@ -135,6 +200,81 @@ public static bool UpdateVersionFile(GitClient gitClient, string fileContents, s
             return true;
         }
 
+        public static bool UpdateUnrealEngineVersionFile(GitClient client, List<string> versions, string unrealEngineVersionFile)
+        {
+            using (new WorkingDirectoryScope(client.RepositoryPath))
+            {
+                if (!File.Exists(unrealEngineVersionFile))
+                {
+                    throw new InvalidOperationException("Could not update the unreal engine file as the file " +
+                        $"'{unrealEngineVersionFile}' does not exist.");
+                }
+                var originalContents = File.ReadAllText(unrealEngineVersionFile);
+                File.WriteAllLines(unrealEngineVersionFile, versions);
+
+                // If nothing has changed, return false, so we can react to it from the caller.
+                if (File.ReadAllText(unrealEngineVersionFile) == originalContents)
+                {
+                    return false;
+                }
+
+                client.StageFile(unrealEngineVersionFile);
+                return true;
+            }
+        }
+
+        public static bool UpdatePluginFile(GitClient gitClient, string version, string pluginFileName, NLog.Logger Logger)
```

Comment:
Same as before except it now throws if there is no plugin file found and it returns if its actually updated anything